lib/node_modules/@stdlib/repl/lib/editor_actions.js

```diff
@@ -531,17 +532,18 @@ setNonEnumerableReadOnly( EditorActions.prototype, 'beforeKeypress', function be
 		return false;
 	}
 	inputKeys = parseKey( key );
+	isTriggered = false;
 	for ( i = 0; i < EDITOR_ACTIONS.length; i++ ) {
 		actionKeys = this._keybindings[ EDITOR_ACTIONS[ i ] ];
 		if ( !actionKeys ) {
 			continue; // no keybindings configured for the action
 		}
 		if ( matchKeybindings( inputKeys, actionKeys ) ) {
 			this[ EDITOR_ACTIONS[ i ] ]();
-			return true;
+			isTriggered = true;
```

Comment:
What is the rationale for this change, as it doesn't seem to me like a change in behavior?

Reply:
When a user sets a new keybinding using the new listener, and if it clashes with any of the current keybindings, then it might not work the moment they go to try out the newly set keybinding, as we are currently supporting only one action per keybinding. This can seem like a big given the user just set the keybinding and trying it doesn't work, and they don't know about the clash.

So I figured, allowing the keybinding to trigger multiple actions can work. Or we can validate if it's not an existing keybinding and allow them to enter again.

Reply:
Ah, okay. Seems reasonable. Yeah, my guess is that we want to verify that it's not an existing binding. And if it is, let the user know. Having a binding trigger multiple actions seems like not a good idea.

Reply:
But then that would also call for validating the `keybindings` option, and I'd assume that checking if each key object in a keybindings object is unique or not would be high in time complexity. Another way I can think is, just giving a warning with the output, similar to VSCode:

```
Listening for keypresses...

Successfully set the following key bindings for action `clearScreen`.

[
   {
      "name": "a",
      "ctrl": true",
      "shift": false,
      "meta": false
   }
]

Warning! 2 existing actions have this keybinding: `moveLeft`, `tab`.
```

Reply:
Why high in time complexity? It would just be a linear scan, correct? Could wrap as a small utility module `find_keybinding.js` which returns a list of matching actions. If time complexity is high, another way would be to hash key objects and store in a key cache. If present, have a conflict. The latter would require, however, keeping the cache up-to-date.

Reply:
> Why high in time complexity? It would just be a linear scan, correct? Could wrap as a small utility module find_keybinding.js which returns a list of matching actions.

Yes. I meant high for determining if all the keys are unique in keybindings object given during REPL startup. If we don't want to support a keybinding triggering multiple actions, we would have to check that when we validate the `keybindings` object during startup. We can use a hashmap here though, so not a problem I think.

All in all, we can either allow a keybinding to trigger multiple keybindings, and leave users with a warning (similar to VSCode), or strictly validate if each keybinding should trigger one action.

Reply:
Is there a use case for allowing a single keybinding to trigger multiple actions? Just trying to think when a user would actually want this.

Reply:
And do we allow multiple actions to be triggered? Or is the implementation first-come first-action?

Reply:
I can sort of see a use case where a user could want keyboard macro-like behavior, but that seems distinct from setting up keybindings for select editor actions, as done here.

Reply:
Yeah, I don't see a clear use case.

Reply:
Given that I don't think a user will expect peak performance for assigning a keybinding, I think validating to check for duplicates should be fine. As a first pass, I suggest just having a naive implementation which searches for an existing keybinding in the list of keybindings. We can always optimize later with hashing techniques.

Reply:
Searching an existing keybinding is not the problem. But validating the keybinding object during startup becomes one. As the number of keys in the keybindings object can be quite large, (an array of possible keys for just one action, we have to validate if each of those keys for all actions is unique or not).. or maybe not?

Reply:
Yeah, you are right, in the sense that having a keybinding hash (lookup table) is likely the most reasonable approach. Just requires keeping things in sync. Namely, if a new keybinding is added, the hash also needs updating. If a keybinding is removed, the hash needs updating. Etc. You could always create a keybindings class, which internally manages the two datasets (namely, action => keybinding and keybinding => action), and then define a convenient set of methods for querying whether a keybinding already exists. In terms of hash function, the easiest may be to just JSON.stringify a keybinding.

In terms of startup cost, I am less worried about this atm. Launching a REPL is likely to not have stringent user expectations in terms of load time. The cost here is at most milliseconds. If it is more than that, say, a few seconds, just to validate a set of keybindings, then we might have some problems.